app/src/androidTest/java/org/mozilla/tv/firefox/ui/YouTubeNavigationTest.kt

```diff
@@ -0,0 +1,179 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.ui
+
+import android.app.Application
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.tv.firefox.TestDependencyFactory
+import org.mozilla.tv.firefox.ext.toUri
+import org.mozilla.tv.firefox.helpers.AndroidAssetDispatcher
+import org.mozilla.tv.firefox.helpers.CustomPocketFeedStateProvider
+import org.mozilla.tv.firefox.helpers.MainActivityTestRule
+import org.mozilla.tv.firefox.helpers.TestAssetHelper
+import org.mozilla.tv.firefox.pocket.PocketVideoRepo
+import org.mozilla.tv.firefox.pocket.PocketViewModel
+import org.mozilla.tv.firefox.ui.robots.browser
+import org.mozilla.tv.firefox.ui.robots.navigationOverlay
+import org.mozilla.tv.firefox.utils.ServiceLocator
+
+/**
+ * A test for YouTube navigation including:
+ * - Going back with hardware back acts as expected from YouTube tile and Pocket videos
+ * - D-pad navigation works as expected
+ * - Checking that clicking on the YouTube tile, opening the overlay, waiting for loading to end, then closing the overlay
+ *   does not disrupt navigation
+ */
+class YouTubeNavigationTest {
+    companion object : TestDependencyFactory {
+        private val customPocketFeedStateProvider = CustomPocketFeedStateProvider()
+
+        override fun createServiceLocator(app: Application) = object : ServiceLocator(app) {
+            override val pocketRepo = customPocketFeedStateProvider.fakedPocketRepo
+        }
+    }
+
+    @get:Rule val activityTestRule = MainActivityTestRule()
+    private lateinit var server: MockWebServer
+    private lateinit var pages: List<TestAssetHelper.TestAsset>
+
+    /* ktlint-disable no-blank-line-before-rbrace */ // This imposes unreadable grouping.
+
+    @Before
+    fun setup() {
+        server = MockWebServer().apply {
+            setDispatcher(AndroidAssetDispatcher())
+            start()
+        }
+
+        pages = TestAssetHelper.getGenericAssets(server)
+
+        val mockedState = PocketVideoRepo.FeedState.LoadComplete(listOf(
+            PocketViewModel.FeedItem.Video(
+                id = 27587,
+                title = "How a Master Pastry Chef Uses Architecture to Make Sky High Pastries",
+                url = "https://www.youtube.com/tv#/watch/video/idle?v=953Qt4FnAcU",
+                thumbnailURL = "https://img-getpocket.cdn.mozilla.net/direct?url=http%3A%2F%2Fimg.youtube.com%2Fvi%2F953Qt4FnAcU%2Fmaxresdefault.jpg&resize=w450",
+                popularitySortId = 20,
+                authors = "Eater"
+            )
+        ))
+
+        customPocketFeedStateProvider.fakedPocketRepoState.onNext(mockedState)
+    }
+
+    @Test
+    fun youtubeNavigationTest() {
+        val youtubeUrl = "youtube.com/tv"
+        val youtubeFinalUrl = "youtube.com/tv#/surface?c=FEtopics&resume"
```

Comment:
Nit: might be more clear to rename these `youTubeUrlBeforeRedirects` and `youTubeUrlAfterRedirects`